runtime/tests/checker/interface_test.go

```diff
@@ -2998,6 +2998,30 @@ func TestCheckInterfaceInheritance(t *testing.T) {
 		assert.Equal(t, "Foo", memberConflictError.ConflictingInterfaceType.QualifiedIdentifier())
 	})
 
+	t.Run("duplicate methods mismatching entitlements", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+			entitlement X
+
+            struct interface Foo {
+                access(X) fun hello()
+            }
+
+            struct interface Bar: Foo {
+                pub fun hello(): String
```

Comment:
Could be mismatching also because of the different return types. So maybe keep the rest of the function signature identical.
```suggestion
                pub fun hello()
```